### DISCUSSION TITLE
Timestamps - X-axis

### DISCUSSION THREAD

**Discussion by gand07** [Daveiano/weewx-wdc#199] [Q&A]
I have started to look at the possibilities for graphs and I have tried to make a combined graph, Rain Total / Cumulative. As you can see from the screenshot, the time indication is different from the one above. What parameter should I use to get 4 timestamps. I don't think I can see anything in WiKI.

The definitions are here:
[[diagrams]]
        #[[[Rounding]]]
            #barometer = 3

        [[[[rain_cum_sum]]]]
            label = 'Rain Total / Cumulative'
            yScaleMin = 0
            [[[[[obs]]]]]
                [[[[[[rain_sum]]]]]]
                    observation = "rain"
                    aggregate_type = sum
                    # aggregate_interval = 7200 # 2 hours
                [[[[[[rain_cum]]]]]]
                    observation = "rain"
                    color = "#ffc000"
                    type = line
                    aggregate_type = cumulative
                    # aggregate_interval = 7200 # 2 hours
                    reset = midnight
                    curve = linear

[[[day]]]
            aggregate_interval = 1800 # 30 minutes
            bottom_date_time_format = "%H:%M"
	    tooltip_date_time_format = "%d-%m-%y %H:%M"
            [[[[observations]]]]
                [[[[[tempdew]]]]]
                [[[[[rainRate]]]]]
		[[[[[rain_cum_sum]]]]]
                [[[[[wind]]]]]
                [[[[[windDir]]]]]
                [[[[[windRose]]]]]
				[[[[[windchill_heatindex]]]]]
                [[[[[outHumidity]]]]]
                [[[[[barometer]]]]] 
                [[[[[UV]]]]]
                [[[[[radiation]]]]]
                [[[[[cloudbase]]]]]
                #[[[[[appTemp]]]]]

Thanks for looking
![Graph](https://github.com/Daveiano/weewx-wdc/assets/47734099/bf510062-ecf8-44ac-957c-0c9cff625b01)

**Comment by Daveiano** [comment]
The charts use the first Observation for the xAxis legend (if it is a combined chart).

So in this case it's a combined chart of rain_sum and rain_cum. The first one is rain_sum, which is a bar graph. For bar graphs, there is a date label shown on the xAxis for EVERY bar - that's because there are so many items on your xAxis.

You should be able to get the display from above if you swap the two observations:

```
[[[[rain_cum_sum]]]]
        label = 'Rain Total / Cumulative'
        yScaleMin = 0
        [[[[[obs]]]]]
            [[[[[[rain_cum]]]]]]
                observation = "rain"
                color = "#ffc000"
                type = line
                aggregate_type = cumulative
                # aggregate_interval = 7200 # 2 hours
                reset = midnight
                curve = linear
            [[[[[[rain_sum]]]]]]
                observation = "rain"
                aggregate_type = sum
                # aggregate_interval = 7200 # 2 hours
```

I will need to look into this and how the default WeeWX Season skin handles this - the current behaviour seems a little odd, I guess!?
